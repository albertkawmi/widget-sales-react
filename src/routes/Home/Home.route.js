import React from 'react'
import { Link } from 'react-router-dom'

export const Home = () => (
  <div className="home">
    <h2>Welcome</h2>
    <p>What would you like to do?</p>
    <Link to="/clients">
      <button className="mr">View all your Clients</button>
    </Link>
    <Link to="/sales">
      <button className="mr">View all your Sales</button>
    </Link>
    <Link to="/fault">
      <button>Actually, I'd like to see an error page instead</button>
    </Link>
  </div>
)
