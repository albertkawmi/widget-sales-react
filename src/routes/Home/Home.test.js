/* eslint-env jest */
import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';
import Home from './Home.route'

Enzyme.configure({ adapter: new Adapter() });

const homeRoute = Enzyme.shallow(
  <Router>
    <Home />
  </Router>
)

describe('<Home />', () => {
  it('renders without crashing', () => {
    expect(homeRoute).toBeTruthy()
  })

  it('has not been changed unintentionally', () => {
    expect(homeRoute).toMatchSnapshot()
  })
})
