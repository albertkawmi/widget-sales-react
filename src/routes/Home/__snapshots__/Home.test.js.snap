// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders the home page 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Welcome
</h2>,
        <p>
          What would you like to do?
</p>,
        <Link
          replace={false}
          to="/clients"
>
          <button
                    className="mr"
          >
                    View all your Clients
          </button>
</Link>,
        <Link
          replace={false}
          to="/sales"
>
          <button
                    className="mr"
          >
                    View all your Sales
          </button>
</Link>,
        <Link
          replace={false}
          to="/fault"
>
          <button>
                    Actually, I'd like to see an error page instead
          </button>
</Link>,
      ],
      "className": "home",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Welcome",
        },
        "ref": null,
        "rendered": "Welcome",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "What would you like to do?",
        },
        "ref": null,
        "rendered": "What would you like to do?",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <button
            className="mr"
>
            View all your Clients
</button>,
          "replace": false,
          "to": "/clients",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "View all your Clients",
            "className": "mr",
          },
          "ref": null,
          "rendered": "View all your Clients",
          "type": "button",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <button
            className="mr"
>
            View all your Sales
</button>,
          "replace": false,
          "to": "/sales",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "View all your Sales",
            "className": "mr",
          },
          "ref": null,
          "rendered": "View all your Sales",
          "type": "button",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <button>
            Actually, I'd like to see an error page instead
</button>,
          "replace": false,
          "to": "/fault",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Actually, I'd like to see an error page instead",
          },
          "ref": null,
          "rendered": "Actually, I'd like to see an error page instead",
          "type": "button",
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Welcome
</h2>,
          <p>
            What would you like to do?
</p>,
          <Link
            replace={false}
            to="/clients"
>
            <button
                        className="mr"
            >
                        View all your Clients
            </button>
</Link>,
          <Link
            replace={false}
            to="/sales"
>
            <button
                        className="mr"
            >
                        View all your Sales
            </button>
</Link>,
          <Link
            replace={false}
            to="/fault"
>
            <button>
                        Actually, I'd like to see an error page instead
            </button>
</Link>,
        ],
        "className": "home",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Welcome",
          },
          "ref": null,
          "rendered": "Welcome",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "What would you like to do?",
          },
          "ref": null,
          "rendered": "What would you like to do?",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <button
              className="mr"
>
              View all your Clients
</button>,
            "replace": false,
            "to": "/clients",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "View all your Clients",
              "className": "mr",
            },
            "ref": null,
            "rendered": "View all your Clients",
            "type": "button",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <button
              className="mr"
>
              View all your Sales
</button>,
            "replace": false,
            "to": "/sales",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "View all your Sales",
              "className": "mr",
            },
            "ref": null,
            "rendered": "View all your Sales",
            "type": "button",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <button>
              Actually, I'd like to see an error page instead
</button>,
            "replace": false,
            "to": "/fault",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Actually, I'd like to see an error page instead",
            },
            "ref": null,
            "rendered": "Actually, I'd like to see an error page instead",
            "type": "button",
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
